core/api/src/servers/authentication/index.ts

```diff
@@ -307,6 +295,37 @@ authRouter.post("/phone/code", async (req: Request, res: Response) => {
   })
 })
 
+authRouter.post("/phone/code-appcheck", async (req: Request, res: Response) => {
+  const appcheckJti = req.headers["x-appcheck-jti"]
+  if (!appcheckJti || typeof appcheckJti !== "string" || appcheckJti === "") {
+    return res.status(400).send({ error: "missing or invalid appcheck jti" })
+  }
+
+  const ip = req.originalIp
```

Comment:
should not implement the same code used in setGqlContext?

Reply:
`setGqlContext` is for graphql routes. there are already 2 different setGqlContext (one for public and one for admin routes). so I'm not sure it's relevant to use/reuse code from setGqlContext because here it's a different route (`/auth` router) with a different context, but maybe I'm missing something?